base/sort.jl

```diff
@@ -349,6 +359,38 @@ function sort!(v::AbstractVector, lo::Int, hi::Int, a::MergeSortAlg, o::Ordering
     return v
 end
 
+function sort!(v::AbstractVector, lo::Int, hi::Int, a::PartialQuickSort,
+               o::Ordering)
+    k = a.k
+    while lo < hi
+        hi-lo <= SMALL_THRESHOLD && return sort!(v, lo, hi, SMALL_ALGORITHM, o)
+        pivot = v[(lo+hi)>>>1]
```

Comment:
Mentioned in the general comments, but `QuickSort` has an improved pivot selection (with minimal extra cost and the side benefit of more balanced sorting), which should be extracted to a function and called from here.